_RFCs/0007-deprecating-packages.md

```diff
@@ -0,0 +1,59 @@
+j- Change Name: deprecate_unsupported_software_plans
+- Start Date: 2018-05-07
+- RFC PR: (leave this empty)
+- Core-Plans Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+When a piece of software that is packaged with a Habitat plan is no longer supported by the manufacturer of that software, we need a process to deprecate that plan.
+
+Deprecating a plan does NOT involve removing it from builder. Rather, we must indicate to users that the plan will no longer be updated by the Habitat team except in cases of security vulnerabilities.
+
+This RFC proposes a process to:
+* Deprecate a plan
+* Communicate that deprecation notice to all users
+* Establish a process for handling emergency updates - like serious regression bugs
+
+# Motivation
+[motivation]: #motivation
+
+We need a way to indicate that a plan is deprecated without removing the plan itself from Builder (as this could potentially cause breakages for anyone who is still dependent on it), and to redirect users to updated plans that are still supported by the manufacturer.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We currently have plans that package software which the author/manufacturer has declared EOL or otherwise no longer updated. This includes, but may not be limited to:
+* [core/ruby22](https://github.com/habitat-sh/core-plans/tree/master/ruby22) (The Ruby team has indicated [here](https://www.ruby-lang.org/en/news/2018/03/28/ruby-2-2-10-released/) that Ruby 2.2.x security maintenance ended in March 2018. Additionally, we discovered during the latest base plans refresh that Ruby 2.2.x does not work with newer versions of glibc and gcc. While we can pin it to older versions of glibc and gcc, this will introduce potential dependency issues when used with software that was created with newer versions of those base packages)
+* [core/jre9](https://github.com/habitat-sh/core-plans/tree/master/jre9)(Oracle has indicated [here](http://www.oracle.com/technetwork/java/javase/downloads/jre9-downloads-3848532.html) that Java SE 9 has reached end of support and users should upgrade to Java SE 10. Additionally, Oracle's download links to JRE 9 and JDK 9 have been removed. It should be noted that Oracle DOES still support JRE 8 and JDK 8.)
+* [core/jdk9](https://github.com/habitat-sh/core-plans/blob/master/jdk9/plan.sh)(see explanation for core/jre9)
+
+When we become aware that a piece of software we package in core plans has been EOL'd or the author/manufacturer otherwise stops supporting it, we will:
+* Open an RFC explaining why the plan must be deprecated
+* When that RFC is approved (though the normal RFC approval process), we must
+1) Delist the package in builder by setting it to "private" (that will make it invisible to anyone outside of the core org, as mentioned [in this issue](https://github.com/habitat-sh/builder/issues/18)
+2) Remove the core plan from the [core plans repo](https://github.com/habitat-sh/core-plans)
```

Comment:
This implies that any packages depending on the deprecated one will not be usable after a base refresh. Thus, I think packages in the core origin that depend on the deprecated one should also be deprecated.

Also, this might prevent people from using newer hab versions (with `hab-backline` updated) because those deprecated packages won't be compatible with the new base as they won't be rebuilt. 

I think we should keep the plan, mark it deprecated in its plan sources, prevent upgrades unless strictly required, still private in builder, but let hab use the newer releases (that would have the same version, but with updated dependencies). 

the alternative, IMHO, is that deprecated would actually mean "until the next base refresh comes, you can still use it, after, you're out of luck" and it _may_ be a bit too fast for our users :) (Think java, the versions have a tendency to get used way after the EOL, even if it might change, not sure it will happen soon)

Reply:
You are super right. Do what @rsertelon says.